contracts/InputSettlerXCMEscrow.sol

```diff
@@ -0,0 +1,307 @@
+pragma solidity ^0.8.26;
+import {InputSettlerPurchase} from "oif/input/InputSettlerPurchase.sol";
+import {InputSettlerEscrow} from "oif/input/escrow/InputSettlerEscrow.sol";
+import {IInputSettlerEscrow} from "oif/interfaces/IInputSettlerEscrow.sol";
+import {InputSettlerBase} from "oif/input/InputSettlerBase.sol";
+import {EIP712} from "openzeppelin/utils/cryptography/EIP712.sol";
+import {Ownable} from "openzeppelin/access/Ownable.sol";
+import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
+import {StandardOrder, StandardOrderType} from "oif/input/types/StandardOrderType.sol";
+import {MandateOutput} from "oif/input/types/MandateOutputType.sol";
+import {OrderPurchase} from "oif/input/types/OrderPurchaseType.sol";
+import {LibAddress} from "oif/libs/LibAddress.sol";
+import {IXcm} from "./interfaces/IXcm.sol";
+import {ILibrary} from "./interfaces/ILibrary.sol";
+
+contract InputSettlerXCMEscrow is
+    InputSettlerPurchase,
+    IInputSettlerEscrow,
+    Ownable
+{
+    using StandardOrderType for StandardOrder;
+    using LibAddress for uint256;
+    using LibAddress for bytes32;
+    address inkLibrary;
+    address xcmPrecompile;
+    address baseSettler;
```

Comment:
```suggestion
    address public immutable inkLibrary;
    address public immutable xcmPrecompile;
    address public immutable baseSettler;
```
so can be queried by off-chain components without scraping deployment logs or reading byte code